packages/adapter-neon/src/neon.ts

```diff
@@ -154,6 +154,13 @@ class NeonTransaction extends NeonWsQueryable<neon.PoolClient> implements Transa
     super(client)
   }
 
+  async begin(): Promise<Result<void>> {
+    debug(`[js::begin]`)
+
+    this.client.release()
+    return Promise.resolve(ok(undefined))
+  }
```

Comment:
I'm not sure I understand this. Why does `begin` close the transaction?

Reply:
This is a mistake, I think maybe a copypasta from the `commit` implementation 🤔 
I'll remove the release line.